core/server/worker/src/main/java/alluxio/worker/block/RemoteBlockReader.java

```diff
@@ -29,7 +30,7 @@
 /**
  * Reads a block from a remote worker node. This should only be used for reading entire blocks.
  */
-public class RemoteBlockReader implements BlockReader {
+public class RemoteBlockReader extends AbstractBlockClient implements BlockReader {
```

Comment:
I think the code would be cleaner if `BlockReader` and `BlockWriter` were an abstract classes which extended `AbstractBlockClient` so that is wouldn't require both an `extends` and `implements` here. Then we could just have the default constructor for the Reader/writer set its type to `READER/WRITER` and you wouldn't need `super(READER)/super(WRITER)` everywhere since it would be taken care of by the class def extending `BlockReader` or `BlockWriter`

Reply:
I didn't want to change the existing class names.
However, I think it's OK to make them abstract without stating in their class name...